addons/html_builder/static/src/builder/plugins/move/move_plugin.js

```diff
@@ -0,0 +1,192 @@
+import { Plugin } from "@html_editor/plugin";
+import {
+    addMobileOrders,
+    removeMobileOrders,
+} from "@html_builder/builder/utils/column_layout_utils";
+import { isMobileView } from "@html_builder/builder/utils/utils";
+
+// TODO find where to define the selectors so they are not duplicated across files
+const moveUpOrDown = {
+    selector: [
+        "section",
+        ".s_accordion .accordion-item",
+        ".s_showcase .row .row:not(.s_col_no_resize) > div",
+        ".s_hr",
+        // In snippets files
+        ".s_pricelist_boxed_item",
+        ".s_pricelist_cafe_item",
+        ".s_product_catalog_dish",
+        ".s_timeline_list_row",
+        ".s_timeline_row",
+    ].join(", "),
+};
+
+const moveLeftOrRight = {
+    selector: [
+        ".row:not(.s_col_no_resize) > div",
+        ".nav-item",
+        ".s_timeline_card", // timeline TODO custom function, other plugin ?
+    ].join(", "),
+    exclude: ".s_showcase .row .row > div",
+};
+
+export class MovePlugin extends Plugin {
+    static id = "move";
+    static dependencies = ["overlay", "history"];
+    resources = {
+        has_overlay_buttons: this.isTargetMovable.bind(this),
+        get_overlay_buttons: this.getActiveOverlayButtons.bind(this),
+    };
+
+    setup() {
+        this.target = null;
+        this.state = {
+            isMobileView: false,
+            isGridItem: true,
+        };
+    }
+
+    getActiveOverlayButtons({ target, callback }) {
+        const buttons = [];
+        if (this.isTargetMovable(target)) {
+            this.target = target;
```

Comment:
why do you need to put the target on this. Maybe just pass it at your callback ?

Reply:
I need to save the target somewhere because we need to know it when the button will be clicked later :slightly_smiling_face: I find it logical that as long as the overlay is open, the plugin has a target on which it applies all its function :slightly_smiling_face: 
But I indeed forgot to set it to null when closing the overlay :grimacing: I will fix that :+1: